python/pyspark/errors/error_classes.py

```diff
@@ -742,6 +742,11 @@
       "Only allows <arg_name> to be a path without scheme, and Spark Driver should use the default scheme to determine the destination file system."
     ]
   },
+  "NULLABILITY_CHECK_FAILED": {
```

Comment:
Can we name it like `NOT_NONE` to be consistent with others like`NOT_STR`, `NOT_STRUCT`?

Reply:
+1， BTW, do we plan to consolidate those `NOT_STR`/`NOT_STRUCT` error classes.

Reply:
We have a plan to refine the error classes, but not sure which way is the best:

1. We can create one representing main error class and make these similar error classes into sub-error classes under it.
2. We can consolidate them all and make the type name as a message parameter.

Reply:
I personally prefer option 2.

Reply:
Yeah although I have no strong opinion, but I also bit more prefer 2.

The only thing I've been concerning is that it may harm readability in some places, but I believe it would not that much.

Reply:
Will create a ticket for it